test/ng/directive/inputSpec.js

```diff
@@ -1918,6 +1918,83 @@ describe('input', function() {
     });
   });
 
+  ['month', 'week', 'time', 'date', 'datetime-local'].forEach(function(inputType) {
+    if (jqLite('<input type="' + inputType + '">').attr('type') !== inputType) {
```

Comment:
This fails to detect the supporting browsers. You should use `.prop()` instead.
The `type` attribute will always have the requested type, but the `type` property will be set to `test` if the browser doesn't support the type.